### PR TITLE
[DEV-1560] integrate logic package

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,6 +88,9 @@
     },
     "collectCoverageFrom": [
       "**/*.(t|j)s"
+    ],
+    "transformIgnorePatterns": [
+      "node_modules/(?!@dfx)"
     ]
```
Comment (davidleomay) [inline package.json:94]: What is this for?

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,22 @@
...
+
+function AppWrapper(): JSX.Element {
+  const { signMessage, connect, address, blockchain, isConnected } = useWalletContext();
+
+  return (
+    <DfxContextProvider api={{ signMessage, connect }} data={{ address, blockchain, isConnected }}>
+      <Main />
+    </DfxContextProvider>
   );
 }
```
Comment (davidleomay) [inline src/App.tsx:21]: Nice 😉

File: src/components/tabs/buy-tab-content/buy.overview.tsx [hunk 3]
```diff
@@ -14,26 +14,51 @@ export function BuyTabContentOverview({ onAssetClicked }: BuyTabContentOverviewP
...
+    [Blockchain.OPTIMISM]: 3,
+    [Blockchain.POLYGON]: 4,
+    [Blockchain.DEFICHAIN]: Infinity,
+    [Blockchain.BITCOIN]: Infinity,
+    [Blockchain.CARDANO]: Infinity,
+  };
+
+  function orderByBlockchain(a: [Blockchain, Asset[]], b: [Blockchain, Asset[]]): number {
+    return blockchainToOrder[a[0]] - blockchainToOrder[b[0]];
+  }
```
Comment (davidleomay) [inline src/components/tabs/buy-tab-content/buy.overview.tsx:38]: In the current exchange, the blockchain ordering can be done in DB (by setting the sort order on the assets). Is this fixed blockchain order required?

File: src/components/tabs/sell.tab.tsx [hunk 4]
```diff
@@ -24,6 +18,15 @@ import {
...
 } from '@dfx.swiss/react-components';
+import {
+  Asset,
+  Blockchain,
+  useAssetContext,
+  useAuthContext,
+  useSessionContext,
+  useUserContext,
+} from '@dfx.swiss/react';
+import { AssetType } from '@dfx.swiss/react/dist/definitions/asset';
```
Comment (davidleomay) [inline src/components/tabs/sell.tab.tsx:29]: Why is `AssetType` import separated from the rest? 🤔

File: package.json (hunk 1 above)
Comment (Krysh90) [reply package.json:94]: we are importing `@dfx.swiss/react` into test files, this led to building issues which were resolved by adding this line

File: src/App.tsx (hunk 2 above)
Comment (Krysh90) [reply src/App.tsx:21]: I am not quite sure 😓  But we need to wrap them additionally to have access to values of `WalletContext`. This was the only approach which worked

File: src/components/tabs/buy-tab-content/buy.overview.tsx (hunk 3 above)
Comment (Krysh90) [reply src/components/tabs/buy-tab-content/buy.overview.tsx:38]: AFAIK it was based on enum order of exchanges Blockchain enum and within a blockchain you could use sort order. But could be wrong. At least the order was different between DEV and PRD and therefore I have changed it to look like PRD.

File: src/components/tabs/sell.tab.tsx (hunk 4 above)
Comment (Krysh90) [reply src/components/tabs/sell.tab.tsx:29]: good point. because it was not correctly exporting in the package.

**Review by davidleomay — APPROVED** [review]

**Pull request merged**